File: MineAdmin-web/src/modules/base/api/ssoLoginLog.ts
import type { PageList, ResponseStruct } from '#/global'

export interface SSOLoginLogVo {
  id: number
  user_id?: number
  username?: string
  provider_id: number
  provider_name?: string
  provider_display_name?: string
  provider_type?: string
  provider_scene?: string
  binding_id?: number
  sso_user_id?: string
  sso_email?: string
  status: number
  failure_reason?: string
  ip?: string
  user_agent?: string
  device_type?: string
  login_at?: string
}

export interface SSOLoginLogSearchVo {
  page?: number
  page_size?: number
  user_id?: number
  username?: string
  provider_id?: number
  provider_name?: string
  sso_user_id?: string
  sso_email?: string
  status?: number
  start_date?: string
  end_date?: string
}

export interface SSOProviderLogStatVo {
  provider_id: number
  provider_name?: string
  provider_display_name?: string
  total: number
  success_count: number
  fail_count: number
}

export interface SSOLoginStatsVo {
  total: number
  success_count: number
  fail_count: number
  success_rate: number
  providers: SSOProviderLogStatVo[]
}

export function page(data: SSOLoginLogSearchVo): Promise<ResponseStruct<PageList<SSOLoginLogVo>>> {
  return useHttp().get('/admin/sso-login-log/list', { params: data })
}

export function stats(data: SSOLoginLogSearchVo = {}): Promise<ResponseStruct<SSOLoginStatsVo>> {
  return useHttp().get('/admin/sso-login-log/stats', { params: data })
}

File: app/models/sso_login_log.go
package models

import "time"

type SSOLoginLog struct {
	ID            uint64    `gorm:"column:id;primaryKey" json:"id"`
	UserID        uint64    `gorm:"column:user_id" json:"user_id"`
	ProviderID    uint64    `gorm:"column:provider_id" json:"provider_id"`
	BindingID     uint64    `gorm:"column:binding_id" json:"binding_id"`
	SSOUserID     string    `gorm:"column:sso_user_id" json:"sso_user_id"`
	SSOEmail      string    `gorm:"column:sso_email" json:"sso_email"`
	Status        int16     `gorm:"column:status" json:"status"`
	FailureReason string    `gorm:"column:failure_reason" json:"failure_reason"`
	IP            string    `gorm:"column:ip" json:"ip"`
	UserAgent     string    `gorm:"column:user_agent" json:"user_agent"`
	DeviceType    string    `gorm:"column:device_type" json:"device_type"`
	LoginAt       time.Time `gorm:"column:login_at" json:"login_at"`
}

func (SSOLoginLog) TableName() string {
	return "sso_login_log"
}
